File: src/csharp/api/Models/PatientOrder.cs
using Example.Api.Enums;

namespace Example.Api.Models;

/// <summary>
/// Represents an order placed by a patient.
/// </summary>
public record PatientOrder : BaseEntity
{
    /// <summary>
    /// Gets the unique identifier for the order.
    /// </summary>
    public long Id { get; set; }

    /// <summary>
    /// Gets the instructions associated with the order.
    /// </summary>
    public string? Instructions { get; set; }

    /// <summary>
    /// The next visit date for the order.
    /// </summary>
    /// <value></value>
    public DateTimeOffset? NextVisitDate { get; set; }

    /// <summary>
    /// The start date of the order.
    /// </summary>
    /// <value></value>
    public DateTimeOffset? StartDate { get; set; }

    /// <summary>
    /// The end date of the order.
    /// </summary>
    /// <value></value>
    public DateTimeOffset? EndDate { get; set; }

    /// <summary>
    /// The type of the order.
    /// </summary>
    /// <value></value>
    public OrderType Type { get; set; }

    /// <summary>
    /// The status of the order.
    /// </summary>
    /// <value></value>
    public OrderStatus Status { get; set; }

    /// <summary>
    /// The date when the order was dispensed.
    /// </summary>
    /// <value></value>
    public DateTimeOffset? DispensedDate { get; set; }

    /// <summary>
    /// Gets the identifier of the patient who placed the order.
    /// </summary>
    public long PatientId { get; set; }

    /// <summary>
    /// Gets or sets the patient associated with the order.
    /// </summary>
    public Patient? Patient { get; set; }

    /// <summary>
    /// Gets the list of prescriptions associated with the order.
    /// </summary>
    /// <returns></returns>
    public ICollection<Prescription> Prescriptions { get; set; } = [];
}

File: src/python/app/entities/patient_order.py
"""Order entity model."""

from __future__ import annotations
from datetime import datetime
from typing import List, Optional, TYPE_CHECKING

from app.entities.base import Base
from app.entities.base_entity import BaseEntity
from sqlalchemy import DateTime, ForeignKey, String
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.enums.order import OrderType, OrderStatus
from app.db.types import IntEnumType

if TYPE_CHECKING:
    from app.entities.patient import Patient
    from app.entities.prescription import Prescription


class PatientOrder(Base, BaseEntity):
    """Order model."""

    __tablename__ = "patient_order"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    instructions: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)
    next_visit_date: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True)
    start_date: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True)
    end_date: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True)
    dispensed_date: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True)
    type: Mapped[OrderType] = mapped_column(IntEnumType(OrderType), nullable=False, default=OrderType.Prescription)
    status: Mapped[OrderStatus] = mapped_column(IntEnumType(OrderStatus), nullable=False, default=OrderStatus.Created)

    patient_id: Mapped[int] = mapped_column(ForeignKey("patient.id", ondelete="CASCADE"), nullable=False)

    # Relationships
    patient: Mapped["Patient"] = relationship("Patient", back_populates="orders")
    prescriptions: Mapped[List["Prescription"]] = relationship(
        "Prescription", back_populates="order", cascade="all, delete-orphan"
    )
